typewise-alert.c

```diff
@@ -1,52 +1,51 @@
 #include "typewise-alert.h"
 #include <stdio.h>
 
+TemperatureLimits tempLimitArray[3] = {{PASSIVE_COOLING_LOW_LIMIT, PASSIVE_COOLING_HIGH_LIMIT}
+                                          , {HI_ACTIVE_COOLING_LOW_LIMIT, HI_ACTIVE_COOLING_HIGH_LIMIT}
+                                            , {MED_ACTIVE_COOLING_LOW_LIMIT,MED_ACTIVE_COOLING_HIGH_LIMIT}};
+
 BreachType inferBreach(double value, double lowerLimit, double upperLimit) {
   if(value < lowerLimit) {
     return TOO_LOW;
   }
-  if(value > upperLimit) {
+  if(value >= upperLimit) {
     return TOO_HIGH;
   }
   return NORMAL;
 }
 
 BreachType classifyTemperatureBreach(
     CoolingType coolingType, double temperatureInC) {
-  int lowerLimit = 0;
-  int upperLimit = 0;
-  switch(coolingType) {
-    case PASSIVE_COOLING:
-      lowerLimit = 0;
-      upperLimit = 35;
-      break;
-    case HI_ACTIVE_COOLING:
-      lowerLimit = 0;
-      upperLimit = 45;
-      break;
-    case MED_ACTIVE_COOLING:
-      lowerLimit = 0;
-      upperLimit = 40;
-      break;
-  }
-  return inferBreach(temperatureInC, lowerLimit, upperLimit);
+ 
+  return inferBreach(temperatureInC, tempLimitArray[coolingType].lowerLimit, tempLimitArray[coolingType].upperLimit);
 }
 
-void checkAndAlert(
+int checkAndAlert(
     AlertTarget alertTarget, BatteryCharacter batteryChar, double temperatureInC) {
 
   BreachType breachType = classifyTemperatureBreach(
-    batteryChar.coolingType, temperatureInC
-  );
+    batteryChar.coolingType, temperatureInC);
+  
+  return sendAlert(alertTarget, breachType);
+}
+
+int sendAlert (AlertTarget alertTarget, BreachType breachType)
+{
+  if (alertTarget == TO_CONTROLLER)
```

Comment:
The alert propagation is controlled by a if else statement , which means that the production code is aware and hence responsible to send alert to corresponding alert target. This can be improved by using a function pointer that either points to email alerter method or controller alerter method , and the responsibility of which alerter is selected rests with code at a higher level.